frontend/src/common/SortedCollection.ts

```diff
@@ -216,6 +239,18 @@ export class SortedCollection<Item extends { id: number }> {
     this.changeRef += 1;
   }
 
+  // Set a new `Focus`. Any changes to the order of items within the `Focus`
+  // will increment `changeRef`.
+  public setFocus(start: number, end: number) {
+    this.focus = { start, end };
+  }
+
+  // Get the reference to current ordering state of focused items.
+  public ref(): SortedCollection.StateRef {
```

Comment:
I think it might be more idiomatic to use getters and setters here:

so `public setFocus(start: number, end: number)` would become `set focus(start: number, end: number)`
and `public ref(): SortedCollection.StateRef` would become `get ref(): SortedCollection.StateRef`

Then way either of these accessors where called you would omit the parentheses so it would not look like a function is being invoked.

Unclear to me if there is any real benefit to this other than just syntax preference.

Reference: https://www.typescriptlang.org/docs/handbook/classes.html#accessors

Using accessors requires ECMA 5 (ES2015)  or higher, which I think should be fine since we target ES2017 (I am not to familiar with how this all fits into browser compatibility but I assume babel takes care everything.)

Reply:
Ye, `ref` might make sense as a getter, will switch that. I'm not a fan of having a setter that has no corresponding getter (there is no need to check what the focus window currently is, so following YAGNI there is no getter).